dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorPower.java

```diff
@@ -98,7 +99,37 @@ public List<Event> visitStart(Start e) {
         );
 	}
 
-	// =============================================================================================
+    @Override
+    public List<Event> visitInitLock(InitLock e) {
+        return eventSequence(
+                Power.newLwSyncBarrier(),
+                newStore(e.getAddress(), e.getMemValue(), ""));
+    }
+
+    @Override
+    public List<Event> visitLock(Lock e) {
+        Register dummy = e.getThread().newRegister(GlobalSettings.getArchPrecision());
+        Label label = newLabel("FakeDep");
+        // We implement locks as spinlocks which are guaranteed to succeed, i.e. we can
+        // use assumes. The fake control dependency + isync guarantee acquire semantics.
+        return eventSequence(
+                newRMWLoadExclusive(dummy, e.getAddress(), ""),
+                newAssume(new Atom(dummy, COpBin.EQ, IValue.ZERO)),
+                Power.newRMWStoreConditional(e.getAddress(), IValue.ONE, "", true),
```

Comment:
Both use `GlobalSettings.getArchPrecision()`
```
public static IConst ZERO = new IValue(BigInteger.ZERO, getArchPrecision());
```
```
Register dummy = e.getThread().newRegister(GlobalSettings.getArchPrecision());
```

Reply:
It's dangerous because `ZERO` gets initialized only once. If this happens before `GlobalSettings.getArchPrecision()` gets changed (e.g. before the configuration is injected), you get a wrong result.
This issue is probably everywhere in our code, so we can keep it for now. But eventually we will need to make `IValue.ZERO` and `IValue.ONE` functions rather constants.

Reply:
I already did this as part of a WIP branch where I try to fix the problems we have with precision.

Reply:
Then we will defer the issue for now :)